tests/unit/components/App.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { shallowMount } from '@vue/test-utils';
+import App from '@src/components/App.vue';
+
+describe( 'App.vue', () => {
+	it( 'sets the document title on mounted', async () => {
+		shallowMount( App, {
+			props: {
+				assetsPath: 'http://localhost:7072',
+				pageIdentifier: 'test-page',
+				page: { name: 'TestName', template: '<span></span>' },
+				pageTitle: 'test_page_title',
+				pageProps: {},
+				isFullWidth: false,
+				bucketClasses: [],
+				usesContentCards: false,
+			},
+		} );
+
+		expect(document.title).toEqual("{\"key\":\"site_name\",\"pageTitle\":\"{\\\"key\\\":\\\"test_page_title\\\"}\"}");
```

Comment:
Those triple backslashes are when json is encoded multiple times, so you can decode it once and make it a bit more readable:

```suggestion
		expect( JSON.parse( document.title ) ).toStrictEqual( {
			'key': 'site_name',
			'pageTitle': '{"key":"test_page_title"}',
		} );
```